src/main/java/com/autotune/analyzer/recommendations/model/GenericRecommendationModel.java

```diff
@@ -505,54 +505,110 @@ public Map<AnalyzerConstants.RecommendationItem, RecommendationConfigItem> getAc
         for (Map.Entry<Timestamp, IntervalResults> entry : filteredResultsMap.entrySet()) {
             IntervalResults intervalResults = entry.getValue();
 
-            // Skip if accelerator map is null
-            if (null == intervalResults.getAcceleratorMetricResultHashMap())
-                continue;
-
-            // Skip if map is empty
-            if (intervalResults.getAcceleratorMetricResultHashMap().isEmpty())
-                continue;
-
-            isGpuWorkload = true;
-            for (Map.Entry<AnalyzerConstants.MetricName, AcceleratorMetricResult> gpuEntry : intervalResults.getAcceleratorMetricResultHashMap().entrySet()) {
-                AcceleratorMetricResult gpuMetricResult = gpuEntry.getValue();
-
-                // Set Accelerator name
-                if (null == acceleratorModel
-                        && null != gpuMetricResult.getAcceleratorDeviceData().getModelName()
-                        && !gpuMetricResult.getAcceleratorDeviceData().getModelName().isEmpty()
-                        && RecommendationUtils.checkIfModelIsKruizeSupportedMIG(gpuMetricResult.getAcceleratorDeviceData().getModelName())
-                ) {
-                    String obtainedAcceleratorName = RecommendationUtils.getSupportedModelBasedOnModelName(gpuMetricResult.getAcceleratorDeviceData().getModelName());
-
-                    if (null != obtainedAcceleratorName)
-                        acceleratorModel = obtainedAcceleratorName;
+            // Check for accelerator metric map
+            if (null == intervalResults.getAcceleratorMetricResultHashMap() || intervalResults.getAcceleratorMetricResultHashMap().isEmpty()) {
```

Comment:
As mentioned, lets unify both the Remote and Local monitoring usecases from the MetricsResults PoV. We can have this in a subsequent PR.

Reply:
Sure @dinogun